### PR TITLE
Optimize metrics of minute dimensionality persistence, don't save metric of being default value.

### PR DESCRIPTION
**Pull request by wu-sheng**
- [ ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/CHANGES.md).

Optimize metrics of minute dimensionality persistence.
- The value of metrics, which has declaration of the default value and current value equals the default value logically, the whole row wouldn't be pushed into database.
- Add self-observability `metrics_persistence_skipped` metric to measure the number of skipped metrics.
- SLA(percent function), Apdex, AvgLong(MAL mostly) metrics are mostly optimized because of this. Such as a full-failing error(0% successful rate) and unhealthy(apdex=0) are not costing the storage.
- Hour/Day dimensionalities metrics are not changed, as they cover long duration.
- Fix `max` function in OAL doesn't support negative long.

### CODE REVIEW

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/metrics/HavingDefaultValue.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ *
+ * For the declared metrics, the OAP server would skip the persistence of minute dimensionality metrics to reduce
+ * resource costs for the database, when the value of the metric is the default.
+ */
+public interface HavingDefaultValue {
+    /**
+     * @return true if the implementation of this metric has the definition of default value.
+     */
+    default boolean haveDefault() {
+        return false;
```
Comment (hanahmily) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/metrics/HavingDefaultValue.java:33]: It seems all implements return `true`. What's the scenario this has to return `false`?

Comment (wu-sheng) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/metrics/HavingDefaultValue.java:33]: Yes

Comment (wu-sheng) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/metrics/HavingDefaultValue.java:33]: Many others are not having default value, such as heatmap, labelled value(percentile).

Comment (wu-sheng) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/metrics/HavingDefaultValue.java:33]: We have a little chance to optimize the `PercentileFunction` later, but for most labeled value functions, we have to keep them as we need the labels even values are 0.

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/metrics/MaxLongMetrics.java [hunk 2]
```diff
@@ -25,9 +25,6 @@
 import org.apache.skywalking.oap.server.core.analysis.metrics.annotation.SourceFrom;
 import org.apache.skywalking.oap.server.core.storage.annotation.Column;
 
-/**
- *
- **/
 @MetricsFunction(functionName = "max")
 public abstract class MaxLongMetrics extends Metrics implements LongValueHolder {
```
Comment (wankai123) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/metrics/MaxLongMetrics.java:30]: No needs to implement the interface here?

Comment (wu-sheng) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/metrics/MaxLongMetrics.java:30]: Yes, I just added and fixed another bug in this class.

**Review by wankai123 — APPROVED** [review]
LGTM

**Pull request merged**